game-core/src/main/java/games/strategy/net/ServerMessenger.java

```diff
@@ -60,13 +62,57 @@
   private final Map<INode, SocketChannel> nodeToChannel = new ConcurrentHashMap<>();
   private final Map<SocketChannel, INode> channelToNode = new ConcurrentHashMap<>();
 
+  /**
+   * The lobby database if this instance is for use by a lobby or {@code null} if this instance is for use by a game
+   * host.
+   */
+  private final @Nullable Database database;
+
+  /**
+   * Creates a new server messenger for use by a <strong>GAME HOST</strong> using the default object stream factory.
```

Comment:
What do you think of using factory methods over constructors to help complement the commentary?

eg: `ServerMessenger.newGameHostMessenger(...)` and `ServerMessenger.newLobbyMessenger(...)`

Seems like a good fit here for named constructor methods with factory pattern.

Reply:
I considered that but didn't want to have to change a bunch of unrelated code to use the new factory methods.  Upon further review, the vast majority of the constructor calls are in test code, so it might not be that big of a deal after all.  And it would address your previous comment about Javadoc grammar.

I'll look into introducing factory methods later this evening.  If it doesn't feel like it's blowing up this PR, I'll push another commit.

Reply:
I considered that but didn't want to have to change a bunch of unrelated code to use the new factory methods.  Upon further review, the vast majority of the constructor calls are in test code, so it might not be that big of a deal after all.  And it would address your previous comment about Javadoc grammar.

I'll look into introducing factory methods later this evening.  If it doesn't feel like it's blowing up this PR, I'll push another commit.